ensime_shared/launcher.py

```diff
@@ -142,11 +142,17 @@ def generate_classpath(self, scala_version, classpath_file):
         inside_nvim = int(self.vim.eval("has('nvim')"))
         if inside_nvim:
             import tempfile
+            import re
             tmp_dir = tempfile.gettempdir()
             flag_file = "{}/ensime-vim-classpath.flag".format(tmp_dir)
             sbt_cmd += "; echo $? > {}".format(flag_file)
             self.vim.command("echo 'Waiting for generation of classpath...'")
-            self.vim.command("terminal sh \"({} && {})\"".format(cd_cmd, sbt_cmd))
+            if re.match(".+fish$", self.vim.eval("&shell")):
+                sbt_cmd += "; echo $status > {}".format(flag_file)
```

Comment:
The command is also different for both shells
`self.vim.command("terminal {}; and {}".format(cd_cmd, sbt_cmd))`
vs
`self.vim.command("terminal ({} && {})".format(cd_cmd, sbt_cmd))`

`fish` does not accept `&&` and cannot handle parentheses `(cmd)`